contract/src/main/kotlin/io/provenance/scope/loan/contracts/AddENoteModificationContract.kt

```diff
@@ -0,0 +1,45 @@
+package io.provenance.scope.loan.contracts
+
+import io.dartinc.registry.v1beta1.ENote
+import io.provenance.scope.contract.annotations.Function
+import io.provenance.scope.contract.annotations.Input
+import io.provenance.scope.contract.annotations.Participants
+import io.provenance.scope.contract.annotations.Record
+import io.provenance.scope.contract.annotations.ScopeSpecification
+import io.provenance.scope.contract.proto.Specifications.PartyType
+import io.provenance.scope.contract.spec.P8eContract
+import io.provenance.scope.loan.LoanScopeFacts
+import io.provenance.scope.loan.LoanScopeInputs
+import io.provenance.scope.loan.utility.ContractRequirementType
+import io.provenance.scope.loan.utility.documentModificationValidation
+import io.provenance.scope.loan.utility.documentValidation
+import io.provenance.scope.loan.utility.toChecksumMap
+import io.provenance.scope.loan.utility.validateRequirements
+import tech.figure.util.v1beta1.DocumentMetadata
+
+@Participants(roles = [PartyType.OWNER]) // To be invoked by either the servicer or sub-servicer
+@ScopeSpecification(["tech.figure.loan"])
+open class AddENoteModificationContract(
+    @Record(name = LoanScopeFacts.eNote, optional = false) val existingENote: ENote,
+) : P8eContract() {
+
+    @Suppress("Duplicates") /** Identical to [AddENoteAssumptionContract.addAssumption]. */
```

Comment:
If it's that similar, can we just have a type on the proto object to determine if it's an assumption or modification?

Reply:
I think that would have to be a `kv` on the `DocumentMetadata` object. That aside, I think that would potentially restrict our ability to map contract executions to specific event type mappings when processing the event stream?

Reply:
Additionally, it's pretty tough to do any actual DRYing here without first altering `ENote`'s protobuf fields to combine the `modificationsList` and `assumptionsList`.